microsoft-edge/devtools-guide-chromium/memory-problems/dom-leaks.md

```diff
@@ -135,36 +172,99 @@ To identify the culprit node in a detached tree that is causing the entire tree
+<!-- ====================================================================== -->
+## Run GC before getting detached elements, to show only elements that can't be GC'd
+
+In the **Detached Elements** tool, there isn't a single button that both runs GC and then gets detached elements.  To see the needed list of detached elements, showing only the elements that can't be GC'd, you could either: 
+* Click **Collect Garbage**, and then click **Get Detached Elements** (recommended).
+* Click **Get Detached Elements**, click **Collect Garbage**, and then click **Get Detached Elements** again (for demonstration purposes).
+
+The latter approach is only done to demonstrate why you must force GC to run, in order to see only the detached elements that can't be GC'd.  Otherwise, you may end up with a list that shows too many elements - some that can be GC'd, and some that can't.  In practice, you only need to click **Collect Garbage**, and then click **Get Detached Elements**.
+
+Even though some elements might appear as detached at one point in time, you won't know whether they are actually memory leaks until you run GC.  When switching between chat rooms in the demo app, the web page removes the elements that are used to display messages from the DOM.  But switching to a different instance of the `Room` class doesn't delete these elements entirely, so, these elements stay in memory.  Then, when GC runs, GC might delete some of these elements, because such an element is no longer an object in the JavaScript code of the page that references them.  So GC can safely reclaim that memory.
+
+When the user uses the webpage, without DevTools, the user has no control over when GC runs.
+
+
+<!-- copied (& trimmed) section into article "Debug DOM memory leaks ("Detached elements" profiling type)" -->
+<!-- ====================================================================== -->
+## Re-attaching elements
+
+To use the demo webpage, you find detached elements that can't be garbage-collected by the browser, and you identify the JavaScript that's retaining the detached elements.  However, in the context of the demo webpage, it makes sense to retain the list of chat messages, so that if a user switches back to **Room 1**, the message log is preserved.
 
 The following image shows detached elements in the form of messages that are reattached when a user navigates from **Room 2** back to **Room 1**:
 
 ![The detached elements are reattached to the DOM when switching back to Room 1](./dom-leaks-images/reattached.png)
 
-Similarly, a feed in social media might detach elements as users scroll past them, and reattach them to the DOM when users scroll back up. Detached elements aren't always an indication of a memory leak, and memory leaks aren't always caused by detached elements.
+Similarly, a feed in social media might detach elements as users scroll past them, and reattach them to the DOM when users scroll back up.
```

Comment:
result: added:

When the **Detached elements** tool reports detached elements, it's not necessarily a memory leak.  It's up to you to decide what's a memory leak and what's not.  Maybe your app will re-attach those elements later (instead of having to re-create them, which could be slower).

Detaching DOM nodes is a useful approach, as long as you eventually reuse those detached elements (or delete them).  The value of the **Detached elements** tool is, when you suspect a memory leak, you can check whether there are an increasing number of unexpected detached DOM elements being reported by the tool.